### PR TITLE
Better tactics: Cruisers and Fighters now have strafing and dogfighting behavior

### PR DESCRIPTION
**Pull request by BenTheElder**

### CODE REVIEW

File: src/agents/states/Dogfight.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  // In this state, agents attempt to path to an enemy while avoiding combat
+  export class Dogfight extends State {
+    public static DOGFIGHT_RANGE: number = 100;
+
+    public static inDogfightRange(agent: Ship, target: Ship): boolean {
+      return shipDistMinusRadius(agent, target) <= Dogfight.DOGFIGHT_RANGE;
+    }
+    public update(agent: Ship): State {
+      // select a new target when necessary
+      if (!agent.target || agent.target.health <= 0) {
```
Comment (captaincrutches) [inline src/agents/states/Dogfight.ts]: Why don't we just switch back to Idle here - that seems to be our "target picking" state, and it'll restart the process of target acquisition anyway.

Comment (BenTheElder) [inline src/agents/states/Dogfight.ts]: We should, I'll fix that.

**Comment by captaincrutches** [comment]
Looks good except for the duplicate code.  Fix that - and the build, apparently - then ship it.

**Comment by captaincrutches** [comment]
:+1:

**Pull request merged**